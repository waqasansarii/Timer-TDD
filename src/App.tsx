import React from 'react';
import './App.css';
import Timer from './component/timer'


function App() {
  return (
    <div className="App" data-testid="custom-element">
     <Timer  />
    </div>
  );
}

export default App;
