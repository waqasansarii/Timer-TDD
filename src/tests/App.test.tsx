import React from 'react';
import { getByText, render, screen,cleanup } from '@testing-library/react';
import App from '../App';


afterEach(cleanup)
test ('should render App com',()=>{
  const {asFragment} = render(<App />)
  // const element = app.getByTestId('custom-element')
  expect(asFragment()).toMatchSnapshot()
})