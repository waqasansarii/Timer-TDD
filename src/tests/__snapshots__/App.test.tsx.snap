// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App com 1`] = `
<DocumentFragment>
  <div
    class="App"
    data-testid="custom-element"
  >
    <div>
      <div>
        <h1>
          Timer App
        </h1>
      </div>
      <div
        id="clockdiv"
      >
        <div>
          <span
            class="hours"
            data-testid="value"
          >
            00
          </span>
          <div
            class="smalltext"
          >
            Hours
          </div>
        </div>
        <div>
          <span
            class="minutes"
            data-testid="value"
          >
            00
          </span>
          <div
            class="smalltext"
          >
            Minutes
          </div>
        </div>
        <div>
          <span
            class="seconds"
            data-testid="value"
          >
            0
          </span>
          <div
            class="smalltext"
          >
            Seconds
          </div>
        </div>
      </div>
      <div>
        <div
          data-testid="main-div"
        >
          <div
            class="btn_div"
          >
            <button
              class="start"
            >
              start
            </button>
            <button
              class="stop"
            >
              stop
            </button>
            <button
              class="reset"
            >
              reset
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
